docs/get-started/support-new-model.md

```diff
@@ -0,0 +1,548 @@
+# How to support a new model
+
+!!! todo
+    Rename titles and focus on supporting new models
+
+## How kernel optimize a model
+
+### Overview
+
+To optimize a model, kernl use [Torchdynamo](https://github.com/pytorch/torchdynamo) JIT compiler and provide a custom backend where we replace part of the [Torch FX](https://pytorch.org/docs/1.13/fx.html) graph with optimized kernels.
+
+The custom backend is defined in [src/kernl/model_optimization.py](https://github.com/ELS-RD/kernl/blob/74d26c69cd3b12b0097804735ed0555123926503/src/kernl/model_optimization.py#L31)
+
+``` { .py }
+def _compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
+    dynamo_backend_ofi(gm)
+    return cuda_graphs_wrapper(gm, example_inputs, pool=_pool)
+```
+This backend combines two steps:
+
+- First one is to apply the graph replacements
+- Second one is to use [CUDA graphs](https://pytorch.org/blog/accelerating-pytorch-with-cuda-graphs/)
+
+We won't elaborate the second step and focus on the first one that actually does the graph replacements.
+
+### Replace part of the FX graph
+
+First of all, a few word about [Torch FX](https://pytorch.org/docs/1.13/fx.html).
+Torch FX is a torch module to module transformation toolkit. It can trace an torch module (or a function) execution. All the operation are then recorded into a list of nodes that are represented as an FX Graph. From this graph, Torch FX generates python code matching the graph's semantic. Both graph and python code are accessible from the Torch FX GraphModule, which is also a torch module instance. Playing with Torch FX allows us to modify a graph but to stay at the torch module level.
+
+!!! todo
+    Use other diagram that won't bug with mkdocs
+
+``` mermaid
+graph TB
+  A[Torch Module, function] --> B{{Tracer}};
+  F(Torch Module) -.-> C;
+  B --> C[Torch FX GraphModule];
+  C --> D(Torch FX Graph);
+  C --> E(Python Code);
+  D -. Share same semantic .- E;
+```
+
+It's possible to [directly modify the FX graph](https://pytorch.org/docs/1.13/fx.html#direct-graph-manipulation) but for this tutorial we'll rather focus on subgraph rewriting.
+
+To rewrite the Torch FX graph, kernl use the `#!python repace_pattern()` function defined in [src/kernl/utils/extended_matcher.py](https://github.com/ELS-RD/kernl/blob/1463e39aeaaa49535c4a9144c4ab96c6a7a15eca/src/kernl/utils/extended_matcher.py#L337). It's the same function `#!python repace_pattern()` of [Torch FX](https://pytorch.org/docs/1.13/fx.html#subgraph-rewriting-with-replace-pattern) but with some bugfixes (that should be integrated in PyTorch in the future).
```

Comment:
There is no bug fixes, it's more another way to do things which is more focused on our needs. I let you see with @gaetansnl what it is exactly, I will wait your next version to understand myself :-)